### PR TITLE
Tomasz/object key

### PR DESCRIPTION
**Pull request by tmarzec**
## What, How & Why?
This adds the functionality of retrieving Object Keys from realm objects and retrieving objects by their Object Keys.
It is needed by [the realm/realm-flipper-plugin](https://github.com/realm/realm-flipper-plugin).

## ☑️ ToDos

* [ ] 📝 Changelog entry
* [ ] 📝 `Compatibility` label is updated or copied from previous entry
* [x] 🚦 Tests
* [ ] 📦 Updated internal package version in consuming `package.json`s (if updating internal packages)
* [ ] 📱 Check the React Native/other sample apps work if necessary
* [ ] 📝 Public documentation PR created or is not necessary
* [ ] 💥 `Breaking` label has been applied or is not necessary

### CODE REVIEW

**Review by kraenhansen — APPROVED** [review]
Looks legit 👍

**Review by fealebenpae — APPROVED** [review]
💯

File: src/js_realm_object.hpp [hunk 1]
```diff
@@ -295,6 +297,22 @@ std::vector<String<T>> RealmObjectClass<T>::get_property_names(ContextType ctx,
     return names;
 }
 
+template <typename T>
```
Comment (takameyer) [inline src/js_realm_object.hpp:300]: Would be nice to have a comment here explaining the arguments.
For example: https://github.com/realm/realm-js/blob/master/src/js_realm.hpp#L827
We have decided as a team that any new functions should come with comment block headers.

**Review by takameyer — COMMENTED** [review]
Post merge suggestion :)

**Pull request merged**